llvm/lib/CodeGen/MachineRegisterInfo.cpp

```diff
@@ -659,3 +659,11 @@ bool MachineRegisterInfo::isReservedRegUnit(unsigned Unit) const {
   }
   return false;
 }
+
+bool MachineRegisterInfo::isArtificialRegUnit(unsigned Unit) const {
```

Comment:
This doesn't belong in MRI, it does not involve virtual registers.

Reply:
I moved it there because there was already a `isReservedRegUnit`, which I guess also shouldn't live there?
I'll move the function to TRI instead.

Reply:
The set of reserved registers can be changed per function compiled so that needs to live in MRI. The artificial regs are a static property of the target so it can go in TRI.

Reply:
This should go in MCRegisterInfo, it's a constant property of the register. isReservedRegUnit is a stateful property depending on the current MachineFunction, so it should remain here